### PR TITLE
feat(onboarding): skip repo when onboarding PR is up-to-date

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
Skip processing the onboarding branch when onboarding branch and default/base branch has not changed since last run
For this:
- added a new contsant called `REPOSITORY_ONBOARDING_SKIPPED`
- throw an error when the main & onboarding branch SHAs are untouched (the exact location of this needs to be changed)

## Context
The GitHub app has something like 50k onboarding repos. So we need some optimizations to reduce the amount of time spent in processing them.
I think if we skip work if main/onboarding branch SHAs are untouched then it’s quick enough 
_Slack message from_ @rarkins

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
Blocked by: https://github.com/renovatebot/renovate/pull/22452
Once above PR is mergedm the location  of the throw will have to be changed so it doesn't prevent rebase check

File: lib/constants/error-messages.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export const REPOSITORY_NOT_FOUND = 'not-found';
 export const REPOSITORY_NO_PACKAGE_FILES = 'no-package-files';
 export const REPOSITORY_RENAMED = 'renamed';
 export const REPOSITORY_UNINITIATED = 'uninitiated';
+export const REPOSITORY_ONBOARDING_SKIPPED = 'repository-onboarding-skipped'; // skip processing of onboarding repos that are up to date
```
Comment (rarkins) [inline lib/constants/error-messages.ts:36]: Throwing an _error_ for a non-error case seems wrong

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/constants/error-messages.ts (hunk 1 above)
Comment (viceice) [reply lib/constants/error-messages.ts:36]: it's much easier than passing that information all the way up

Comment (rarkins) [reply lib/constants/error-messages.ts:36]: I really don't think a repository being "up to date" should trigger a thrown error, even if it's easy. I would like our code to have less thrown errors

**Pull request closed without merging**